lib/ohai/plugins/linux/network.rb

```diff
@@ -171,6 +171,36 @@ def ethernet_layer_one(iface)
     iface
   end
 
+  # determine ring parameters for the interface using ethtool
+  def ethernet_ring_parameters(iface)
+    return iface unless ethtool_binary = find_ethtool_binary
+    iface.each_key do |tmp_int|
+      next unless iface[tmp_int][:encapsulation] == "Ethernet"
+      so = shell_out("#{ethtool_binary} -g #{tmp_int}")
+      Ohai::Log.debug("Parsing ethtool output: #{so.stdout}")
+      # Nasty, brittle regex to capture the ethtool output
```

Comment:
I can think of two things to make this less brittle:

1. Easy: Don't assume "RX" will always be first, so stick a .* before that in case they add preceeding lines

2. A bit less easy: Parse the output line by line so you don't assume any order:

```
type = nil
lines.each do |line|
  next if line.start_with?('Ring params for')
  next if line.strip.nil?
  if line =~ /Pre-set maximums/
    type = 'max'
    next
  end
  if line =~ /Current hardware settings/
    type = 'current'
    next
  end
  key, val = line.split(/:\s+/)
  iface[tmp_int]["ring_params"]["#{type}_{key.downcase}"] = val
end
```

This will also pull in other settings which will be awesome.